src/mmw/apps/modeling/geoprocessing.py

```diff
@@ -71,6 +72,16 @@ def sjs_submit(host, port, args, data, retry=None):
             except MaxRetriesExceededError:
                 raise Exception('No slots available in Spark JobServer.\n'
                                 'Details = {}'.format(response.text))
+        elif error['result'] == 'context geoprocessing not found':
+            geop_context_url = '{}/contexts?reset=reboot'.format(base_url)
+            context_response = requests.put(geop_context_url)
```

Comment:
Rebooting might be safer, because we're reusing the default context definition, so if we ever change its name we won't have to change it in multiple places. This also protects against other errors that could have happened: it is generally cleaner to reboot the whole thing than to try and piece it back together.